nav2_behavior_tree/plugins/action/remove_in_collision_goals_action.cpp

```diff
@@ -47,23 +48,35 @@ void RemoveInCollisionGoals::on_tick()
   request_->use_footprint = use_footprint_;
 
   for (const auto & goal : input_goals_) {
-    geometry_msgs::msg::Pose2D pose;
-    pose.x = goal.pose.position.x;
-    pose.y = goal.pose.position.y;
-    pose.theta = tf2::getYaw(goal.pose.orientation);
-    request_->poses.push_back(pose);
+    request_->poses.push_back(goal);
   }
 }
 
 BT::NodeStatus RemoveInCollisionGoals::on_completion(
   std::shared_ptr<nav2_msgs::srv::GetCosts::Response> response)
 {
+  for (size_t i = 0; i < input_goals_.size(); ++i) {
+    double yaw = tf2::getYaw(input_goals_[i].pose.orientation);
+    RCLCPP_INFO(
+      node_->get_logger(),
+      "Goal %ld: x: %f, y: %f, yaw: %f, costs: %f", i,
+      input_goals_[i].pose.position.x, input_goals_[i].pose.position.y, yaw, response->costs[i]);
+  }
+
   Goals valid_goal_poses;
   for (size_t i = 0; i < response->costs.size(); ++i) {
-    if (response->costs[i] < cost_threshold_) {
+    if ((response->costs[i] == 255 && !consider_unknown_as_obstacle_) ||
+      response->costs[i] < cost_threshold_)
+    {
       valid_goal_poses.push_back(input_goals_[i]);
     }
   }
+  // Warn if all goals have been removed
+  if (valid_goal_poses.empty()) {
+    RCLCPP_WARN(
```

Comment:
is that a warn or an info? That seems like expected behavior

Reply:
yes it's expected, I just found it useful to know when the output was empty to debug why the planner was throwing an error. But you're right, it can be a simple info